src/foundry/client/pixi/placeables/drawing.d.ts

```diff
@@ -1,204 +1,184 @@
+     * @remarks it's possible this shouldn't be here and is some data model thing
      */
-    controlIcon: null;
+    get layer(): DrawingsLayer;
 
     /**
-     * The inner drawing container
-     * @defaultValue `null`
+     * Each Drawing object provides an interface for a DrawingDocument
      */
-    drawing: PIXI.Container | null;
+    document: DrawingDocument;
 
     /**
-     * The primary drawing shape
-     * @defaultValue `null`
+     * The border frame and resizing handles for the drawing.
      */
-    shape: PIXI.Graphics | null;
+    frame: PIXI.Container;
 
     /**
-     * Text content, if included
+     * A text label that may be displayed as part of the interface layer for the Drawing.
      * @defaultValue `null`
      */
-    text: PIXI.Text | null;
+    text: PreciseText | null;
 
     /**
-     * The Graphics outer frame and handles
+     * The primary drawing shape
      * @defaultValue `null`
      */
-    frame: PIXI.Container | null;
+    shape: DrawingShape;
 
-    /**
-     * Internal timestamp for the previous freehand draw time, to limit sampling
-     * @defaultValue `0`
-     * @internal
-     */
-    protected _drawTime: number;
+    static override embeddedName: "Drawing";
 
-    /**
-     * @defaultValue `0`
-     * @internal
-     */
-    protected _sampleTime: number;
+    static override RENDER_FLAGS: {
+      /** @defaultValue `{propagate: ["refresh"]}` */
+      redraw: RenderFlag<Partial<Drawing.RenderFlags>>;
 
-    /**
-     * Internal flag for the permanent points of the polygon
-     * @defaultValue `foundry.utils.deepClone(this.data.points || [])`
-     * @internal
-     */
-    protected _fixedPoints: Array<[x: number, y: number]>;
+      /** @defaultValue `{propagate: ["refreshState", "refreshShape"], alias: true}` */
+      refresh: RenderFlag<Partial<Drawing.RenderFlags>>;
 
-    static override embeddedName: "Drawing";
+      /** @defaultValue `{propagate: ["refreshFrame"]}` */
+      refreshState: RenderFlag<Partial<Drawing.RenderFlags>>;
+
+      /** @defaultValue `{propagate: ["refreshFrame", "refreshText", "refreshMesh"]}` */
+      refreshShape: RenderFlag<Partial<Drawing.RenderFlags>>;
+
+      /** @defaultValue `{}` */
+      refreshFrame: RenderFlag<Partial<Drawing.RenderFlags>>;
+
+      /** @defaultValue `{}` */
+      refreshText: RenderFlag<Partial<Drawing.RenderFlags>>;
+
+      /** @defaultValue `{}` */
+      refreshMesh: RenderFlag<Partial<Drawing.RenderFlags>>;
+    };
 
     /**
      * The rate at which points are sampled (in milliseconds) during a freehand drawing workflow
      * @defaultValue `75`
      */
     static FREEHAND_SAMPLE_RATE: number;
 
-    override get bounds(): Rectangle;
-
     /**
-     * A Boolean flag for whether or not the Drawing utilizes a tiled texture background
+     * A convenience reference to the possible shape types.
+     * TODO: Replace post-data model with the static enum reference
```

Comment:
What does this mean? Should it refer to `RENDER_FLAGS` or something?

Reply:
Here's the source
```js
  /**
   * A convenience reference to the possible shape types.
   * @enum {string}
   */
  static SHAPE_TYPES = foundry.data.ShapeData.TYPES;
  ```which is referencing
  ```js
class ShapeData extends DataModel {
  static defineSchema() {
    return {
      type: new fields.StringField({required: true, blank: false, choices: Object.values(this.TYPES), initial: "r"}),
      width: new fields.NumberField({required: false, integer: true, min: 0}),
      height: new fields.NumberField({required: false, integer: true, min: 0}),
      radius: new fields.NumberField({required: false, integer: true, positive: true}),
      points: new fields.ArrayField(new fields.NumberField({nullable: false}))
    }
  }

  /**
   * The primitive shape types which are supported
   * @enum {string}
   */
  static TYPES = {
    RECTANGLE: "r",
    CIRCLE: "c",
    ELLIPSE: "e",
    POLYGON: "p"
  }
}
```

So the correct typing would be `ValueOf<(typeof foundry.data.ShapeData)["TYPES"]>`, but that's not available yet

Reply:
Got it! Thanks that makes a lot more sense.